source/Missions/Services/Agents/Handlers/WeaponDropHandler.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Common.Messaging;
+using Common.Network;
+using HarmonyLib;
+using Missions.Services.Agents.Messages;
+using Missions.Services.Agents.Patches;
+using Missions.Services.Network;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TaleWorlds.Core;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Handlers
+{
+    public class WeaponDropHandler
```

Comment:
make everything in this class non-static. Also instantiate these handlers in AutoFac using AutoActivate()